src/fable-library-py/fable_library/date.py

```diff
@@ -60,98 +96,321 @@ def year(d: datetime) -> int:
     return d.year
 
 
+def month(d: datetime) -> int:
+    return d.month
+
+
+def day(d: datetime) -> int:
+    return d.day
+
+
+def hour(d: datetime) -> int:
+    return d.hour
+
+
+def minute(d: datetime) -> int:
+    return d.minute
+
+
+def second(d: datetime) -> int:
+    return d.second
+
+
+def millisecond(d: datetime) -> int:
+    return d.microsecond // 1_000
+
+
+def microsecond(d: datetime) -> int:
+    return d.microsecond
+
+
 def to_universal_time(d: datetime) -> datetime:
     return d.astimezone(timezone.utc)
 
 
+def day_of_week(d: datetime) -> int:
+    return (d.weekday() + 1) % 7
+
+
+def day_of_year(d: datetime) -> int:
+    return d.timetuple().tm_yday
+
+
+def to_short_date_string(date: datetime) -> str:
+    return datetime.strftime(date, "%m/%d/%Y")
+
+
+def to_long_date_string(date: datetime) -> str:
+    return datetime.strftime(date, "%A, %d %B %Y")
+
+
+def to_short_time_string(date: datetime) -> str:
+    return datetime.strftime(date, "%H:%M")
+
+
+def to_long_time_string(date: datetime) -> str:
+    return datetime.strftime(date, "%H:%M:%S")
+
+
+def parse_repeat_token(format: str, pos: int, pattern_char: str) -> int:
+    token_length = 0
+    internal_pos = pos
+    while internal_pos < len(format) and format[internal_pos] == pattern_char:
+        internal_pos += 1
+        token_length += 1
+
+    return token_length
+
+
+# Get the next character at the index of 'pos' in the 'format' string.
+# Return value of -1 means 'pos' is already at the end of the 'format' string.
+# Otherwise, return value is the int value of the next character.
+def parse_next_char(format: str, pos: int) -> int:
+    if pos >= len(format) - 1:
+        return -1
+
+    return ord(format[pos + 1])
+
+
+def parse_quoted_string(format: str, pos: int) -> tuple[str, int]:
+    begin_pos = pos
+    format_length = len(format)
+    # Get the character used to quote the string
+    quote_char = format[pos]
+
+    result = ""
+    found_quote = False
+
+    while pos < format_length:
+        pos += 1
+        current_char = format[pos]
+        if current_char == quote_char:
+            found_quote = True
+            break
+        elif current_char == "\\":
+            if pos < format_length:
+                pos += 1
+                result += format[pos]
+            else:
+                # This means that '\'is the last character in the format string.
+                raise Exception("Invalid string format")
```

Comment:
```suggestion
                raise ValueError("Invalid string format")
```
Ref: https://docs.python.org/3/library/exceptions.html#ValueError